src/main/java/com/laytonsmith/core/events/drivers/BlockEvents.java

```diff
@@ -1519,4 +1519,75 @@ public boolean modifyEvent(String key, Mixed value, BindableEvent event) {
 			return false;
 		}
 	}
+
+	@api
+	public static class block_form extends AbstractEvent {
+
+		@Override
+		public String getName() {
+			return "block_form";
+		}
+
+		@Override
+		public String docs() {
+			return "{newblock: <string match> Type of new block state}"
+					+ "Called when a block is formed or spreads based on world conditions."
+					+ " Cancelling this event block will not be formed."
+					+ "{location: The location the block formed "
+					+ " | block: Type of old block state"
+					+ " | newblock: Type of new block state}"
+					+ "{} "
+					+ "{}";
+		}
+
+		@Override
+		public Driver driver() {
+			return Driver.BLOCK_FORM;
+		}
+
+		@Override
+		public BindableEvent convert(CArray manualObject, Target t) {
+			return null;
+		}
+
+		@Override
+		public Version since() {
+			return MSVersion.V3_3_4;
+		}
+
+		@Override
+		public boolean matches(Map<String, Mixed> prefilter, BindableEvent e) throws PrefilterNonMatchException {
+			if(e instanceof MCBlockFadeEvent) {
+				MCBlockFadeEvent event = (MCBlockFadeEvent) e;
+
+				String newBlock = event.getNewState().getType().getName();
+
+				Prefilters.match(prefilter, "newblock", newBlock, Prefilters.PrefilterType.STRING_MATCH);
+
+				return true;
+			}
+
+			return false;
+		}
+
+		@Override
+		public Map<String, Mixed> evaluate(BindableEvent e) throws EventException {
+			if(!(e instanceof MCBlockFadeEvent)) {
```

Comment:
`MCBlockFadeEvent` -> `MCBlockFormEvent`
On this line, in the exception message and in the code below.

Reply:
Fixed